modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php

```diff
@@ -357,7 +357,12 @@ protected function getAttributeValues(array $variations, $field_name, callable $
    * Ajax callback.
    */
   public static function ajaxRefresh(array $form, FormStateInterface $form_state) {
-    return $form;
+    /** @var \Drupal\Core\Ajax\AjaxResponse $response */
+    $response = \Drupal::service('main_content_renderer.ajax')->renderResponse($form, \Drupal::service('request_stack')->getCurrentRequest(), \Drupal::service('current_route_match'));
+
+    // Allow other modules to add arbitrary AJAX commands on the refresh.
+    \Drupal::moduleHandler()->alter('commerce_product_attributes_refresh', $response, $form, $form_state);
```

Comment:
Ha :P I can read

On Mon, May 2, 2016, 1:37 AM agoradesign <notifications@github.com> wrote:

> In
> modules/product/src/Plugin/Field/FieldWidget/ProductVariationAttributesWidget.php
> <https://github.com/drupalcommerce/commerce/pull/367#discussion_r61707465>
> :
>
> > @@ -357,7 +357,12 @@ protected function getAttributeValues(array $variations, $field_name, callable $
> >     * Ajax callback.
> >     */
> >    public static function ajaxRefresh(array $form, FormStateInterface $form_state) {
> > -    return $form;
> > +    /** @var \Drupal\Core\Ajax\AjaxResponse $response */
> > +    $response = \Drupal::service('main_content_renderer.ajax')->renderResponse($form, \Drupal::service('request_stack')->getCurrentRequest(), \Drupal::service('current_route_match'));
> > +
> > +    // Allow other modules to add arbitrary AJAX commands on the refresh.
> > +    \Drupal::moduleHandler()->alter('commerce_product_attributes_refresh', $response, $form, $form_state);
>
> It's a static function
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/drupalcommerce/commerce/pull/367/files/0c3b15d64902061072457814d17e9d0e3e0f592a#r61707465>
>